### PR TITLE
[MRG] Add refresh token to auth_state if it is available

### PR DESCRIPTION
**Pull request by betatim**
If the OAuth2 server sends a refresh token, add it to the `auth_state`.

### CODE REVIEW

File: oauthenticator/generic.py [hunk 1]
```diff
@@ -134,6 +135,7 @@ def authenticate(self, handler, data=None):
             'name': resp_json.get(self.username_key),
             'auth_state': {
                 'access_token': access_token,
+                'refresh_token': refresh_token,
```
Comment (betatim) [inline oauthenticator/generic.py]: do you prefer the key to always be present (like this) but with `None` as value or only put the key in if the value is not `None`? I am easy either way.

**Comment by betatim** [comment]
There is a whole collection of errors like this:
```python
_______________________ ERROR at teardown of test_auth0 ________________________
    def _close():
        io_loop.clear_current()
>       if (not tornado.ioloop.IOLoop.initialized() or
                io_loop is not tornado.ioloop.IOLoop.instance()):
E               AttributeError: type object 'IOLoop' has no attribute 'initialized'
../../../virtualenv/python3.6.3/lib/python3.6/site-packages/pytest_tornado/plugin.py:136: AttributeError
```

Any ideas what could be causing this? Recent update to tornado?

File: oauthenticator/generic.py (hunk 1 above)
Comment (minrk) [reply oauthenticator/generic.py]: No preference from me

**Comment by betatim** [comment]
🎉

**Pull request merged**